tests/test_local_executor.py

```diff
@@ -507,23 +505,21 @@ def test_running_job_terminated_finalized(docker_cleanup, job, tmp_work_dir):
 
     status = api.terminate(job)
     assert status.state == ExecutorState.ERROR
-    assert api.get_status(job).state == ExecutorState.EXECUTED
+    assert api.get_status(job).state == ExecutorState.ERROR
```

Comment:
if you really don't want to use `ExecutorState.ERROR` in this way, I can probably resurrect the previous iteration that uses the parameterised finalize function (and we just put up with the implementation annoyances)